### ISSUE TITLE
clipPath with image inverts coordinate system.

### ISSUE THREAD

**Issue by blainekasten** [react-native-community/react-native-svg#681]
<details><summary>Code</summary>

This component should be giving a rounded rect with an arrow at the bottom.

```jsx
  const Component = () => (
    <Svg height="56" width="48" viewBox="0 0 48 56">
      <Defs>
        <ClipPath id="tooltip">
          <G stroke="#fff" strokeWidth="1" fill="lime">
            <Rect width="48px" height="48px" rx="5" ry="5" />
            <Path d="M30 47 L25 56 L20 47" />
          </G>
        </ClipPath>
      </Defs>
      <Image
        x="0"
        y="0"
        preserveAspectRatio="xMidYMid slice"
        width="100%"
        height="100%"
        clipPath="url(#tooltip)"
        href={img}
      />
  );
  ```

</details>

<details><summary>Outcome</summary>

As you can see it put the arrow at the top when clipping an image.

![image](https://user-images.githubusercontent.com/2916023/39880142-7ae44c88-5442-11e8-95db-3c5d616c097d.png)

</details>

<details><summary>Outcome with Rect instead of Image</summary>

If I simply replace Image with Rect I get the following

<details><summary>Code</summary>

```jsx
  const Component = () => (
    <Svg height="56" width="48" viewBox="0 0 48 56">
      <Defs>
        <ClipPath id="tooltip">
          <G stroke="#fff" strokeWidth="1" fill="lime">
            <Rect width="48px" height="48px" rx="5" ry="5" />
            <Path d="M30 47 L25 56 L20 47" />
          </G>
        </ClipPath>
      </Defs>
      <Rect width="100%" height="100%" clipPath="url(#tooltip)" fill="blue" />
    </Svg>
  );
```

</details>

![image](https://user-images.githubusercontent.com/2916023/39880227-c18ad44a-5442-11e8-8bab-384fc39c54fd.png)

</details>

**Comment by dapicester** [comment]
I bumped into this too. I can see that only the `y` coordinate gets reversed.

**Comment by mike-marcacci** [comment]
I see an inverted "y" with all `Image`s, with or without a `clipPath` specified. However, the `clipPath` appears to be applied to the image as if it were **not inverted**.

**Comment by mike-marcacci** [comment]
After a cursory look through the code, [this inversion](https://github.com/react-native-community/react-native-svg/blob/f15a2a401bb20aa0671c89ebe001e9bb8a2b854b/ios/Elements/RNSVGImage.m#L126) jumps out as a likely candidate.

I can't dive much deeper today, but figured I'd drop this here in case somebody else his time to pick it up.

**Comment by msand** [comment]
@mike-marcacci good catch, the clipping needed to happen before adjusting image coordinates to screen coordinates (apple likes having the y axis inverted), can you try this fix: https://github.com/react-native-community/react-native-svg/pull/757/commits/f1f0e2f18612236686f1f2deec13553b09cd9958

**Comment by mike-marcacci** [comment]
@msand - this appears to fix the problem with clipPath, but it turns out I was seeing two inverted-y issues. The other one [an inversion of the `Image`'s "y" coordinate](https://github.com/react-native-community/react-native-svg/issues/762) and is fixed by #763.
